### PR TITLE
client: deploy a client node for an external ceph-cluster

### PR DESCRIPTION
**Pull request by guits**
Introduce new role `ceph-config`.
This will give us more flexibility and the possibility to deploy a client node for an external ceph-cluster.

This is related to BZ:
https://bugzilla.redhat.com/show_bug.cgi?id=1469426

Fixes: #1670

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-centos7_cluster

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_dmcrypt

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-dmcrypt_journal

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_cluster

**Comment by colonwq** [comment]
Can you add a task to clean up this (1) temp file? 
1: https://github.com/ceph/ceph-ansible/pull/1749/files#diff-408c9554f34e961d163f68b645ee6da3R16

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-purge_dmcrypt

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-purge_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-purge_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.3-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-xenial_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by leseb** [comment]
CI failures are unrelated and are fixed with https://github.com/ceph/ceph-docker/pull/754

File: roles/ceph-config/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      state: directory
+      owner: "ceph"
+      group: "ceph"
+      mode: "0755"
+    with_items:
+      - /etc/ceph/
+      - /etc/ceph/ceph.d/
+
+  - name: template ceph_conf_overrides
+    local_action: copy content="{{ ceph_conf_overrides }}" dest=/tmp/ceph_conf_overrides_temp
```
Comment (andrewschoen) [inline roles/ceph-config/tasks/main.yml]: Let's use ``fetch_directory`` instead of ``/tmp``.

File: roles/ceph-config/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    run_once: true
+
+  - name: get rendered ceph_conf_overrides
+    local_action: set_fact ceph_conf_overrides_rendered="{{ lookup('template', '/tmp/ceph_conf_overrides_temp') | from_yaml }}"
```
Comment (andrewschoen) [inline roles/ceph-config/tasks/main.yml]: please use ``fetch_directory`` instead of ``/tmp``

File: roles/ceph-config/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    run_once: true
+
+  - name: remove tmp template file for ceph_conf_overrides
+    local_action: file path=/tmp/ceph_conf_overrides_temp state=absent
```
Comment (andrewschoen) [inline roles/ceph-config/tasks/main.yml]: please use ``fetch_directory`` instead of ``/tmp``

File: roles/ceph-config/tasks/main.yml (hunk 1 above)
Comment (leseb) [inline roles/ceph-config/tasks/main.yml]: That was present before, but I can fix it :)

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_dmcrypt

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-purge_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-bluestore_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-bluestore_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-update_docker_cluster

**Comment by leseb** [comment]
The fix for luminous-ansible2.3-update_docker_cluster is here: https://github.com/ceph/ceph-ansible/pull/1795

**Comment by leseb** [comment]
waits for https://github.com/ceph/ceph-ansible/pull/1795

**Pull request merged**